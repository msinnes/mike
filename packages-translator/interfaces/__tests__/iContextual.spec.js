const { isInterface, loadClass } = require('@mike/class');
const Context = require('@mike/translator-classes/Context');

const iContextual = require('../iContextual');

const TestableContext = loadClass(function() {}).extend(Context);

describe('iContextual', () => {
  it('should be an interface', () => {
    expect(isInterface(iContextual)).toBe(true);
  });

  it('should accept a valid config', () => {
    expect(() => {
      iContextual.ensure({
        ContextClass: TestableContext,
      });
    }).not.toThrow();
  });

  it('should throw an error if the config is invalid', () => {
    expect(() => {
      iContextual.ensure({});
    }).toThrowErrorMatchingSnapshot();
  });
});
