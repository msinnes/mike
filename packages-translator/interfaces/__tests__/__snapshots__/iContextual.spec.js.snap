// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iContextual should throw an error if the config is invalid 1`] = `
"ClassError: Runtime Interface Error
ContextClass: Expected ContextClass"
`;
